### PR TITLE
Updating packaging script and separating Shared into its own package

### PR DESCRIPTION
**Pull request by ivelten**
With this PR, I am proposing separating Shared project into its own library. Once done with that, we can go on with splitting this project into smaller, reusable libraries.

### CODE REVIEW

File: build.fsx [hunk 1]
```diff
@@ -203,22 +261,22 @@ Target.create "All" ignore
...
-    ==> "PackServer"
-    ==> "PackClient"
-    ==> "PackMiddleware"
-    ==> "PackAll"
+==> "PackServer"
+==> "PackClient"
+==> "PackMiddleware"
+==> "PackAll"
 
 Target.runOrDefaultWithArguments "All"
```
Comment (xperiandri) [inline build.fsx]: I also tried to format it with Fantomas and it looks not so much readable. @ivelten I should have exclude it from formatting.
Could you to that and only apply your additions

File: src/FSharp.Data.GraphQL.Client/FSharp.Data.GraphQL.Client.fsproj [hunk 2]
```diff
@@ -11,6 +11,11 @@
     <Description>FSharp implementation of Facebook GraphQL query language (Client)</Description>
   </PropertyGroup>
 
+  <ItemGroup>
+    <None Include="..\..\icon.svg" Pack="true" PackagePath="" Visible="False" />
```
Comment (xperiandri) [inline src/FSharp.Data.GraphQL.Client/FSharp.Data.GraphQL.Client.fsproj]: Do we need that icon at all? I thought that NuGet will add it by default. I expected that we will have another icon later.
Is it your intent?

**Review by xperiandri — CHANGES_REQUESTED** [review]

File: src/FSharp.Data.GraphQL.Client/FSharp.Data.GraphQL.Client.fsproj (hunk 2 above)
Comment (ivelten) [reply src/FSharp.Data.GraphQL.Client/FSharp.Data.GraphQL.Client.fsproj]: @xperiandri yes, I saw this entry in one of the fsproj files, and I thought it was already an intent of the team, although the icon file itself was missing. We can certainly remove it at the moment.

File: build.fsx (hunk 1 above)
Comment (ivelten) [reply build.fsx]: Sure, I will do that.

File: src/FSharp.Data.GraphQL.Client/FSharp.Data.GraphQL.Client.fsproj (hunk 2 above)
Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Client/FSharp.Data.GraphQL.Client.fsproj]: I've added some similar stuff to the common `Directory.Build.props`. Check that

**Comment by ivelten** [comment]
@xperiandri just did the requested changes, please take a look.

**Comment by xperiandri** [comment]
https://github.com/fsprojects/FSharp.Data.GraphQL/pull/392 is almost ready.
When it is done, I'll notify you to rebase and update GitHub actions

**Pull request merged**